### PR TITLE
Clean POS code in nbitcoin

### PR DESCRIPTION
**Pull request by dangershony**
The POS code in NBitcoin is not used anymore (its all moved to the fullnode) so this PR deletes that code (and tests)

### CODE REVIEW

File: src/NBitcoin/Block.cs [hunk 1]
```diff
@@ -348,9 +348,6 @@ public void UpdateMerkleRoot()
         /// <returns></returns>
         public bool Check(Consensus consensus)
```
Comment (Aprogiena) [inline src/NBitcoin/Block.cs]: I think you can completely delete this whole method

File: src/NBitcoin/BlockValidator.cs [hunk 2]
```diff
@@ -1,940 +0,0 @@
...
-                var data = ops.ElementAt(1).PushData;
-                if (!ScriptEvaluationContext.IsCompressedOrUncompressedPubKey(data))
-                    return false;
-                return new PubKey(data).Verify(block.GetHash(), new ECDSASignature(block.BlockSignatur.Signature));
-            }
-
-            return false;
-        }
-
-        public static bool IsCanonicalBlockSignature(Block block, bool checkLowS)
```
Comment (Aprogiena) [inline src/NBitcoin/BlockValidator.cs]: This is something that I found in stratisX and it is not explicitly used in C# node, so I was wondering if it is implicitly checked with the current signature verification. But seeing that here, maybe we are missing this check compared to stratisX

**Comment by Aprogiena** [comment]
Looks awesome. I haven't checked if something was broken by this, no tests done, but it looks great. Just please consider my 2 comments.

File: src/NBitcoin/BlockValidator.cs (hunk 2 above)
Comment (dangershony) [reply src/NBitcoin/BlockValidator.cs]: Yeah we have a TODO rule in place for this so I did not forget.
https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosBlockContextRule.cs

That's why I feel its safe to delete from NBitcoin

File: src/NBitcoin/Block.cs (hunk 1 above)
Comment (dangershony) [reply src/NBitcoin/Block.cs]: Ah yeah good spot, its going away.

**Pull request merged**